src/yaml.cr

```diff
@@ -147,4 +148,28 @@ module YAML
   def self.dump(object, io : IO)
     object.to_yaml(io)
   end
+
+  # Determines if the scaler is reserved
+  def self.reserved_scalar(value : String)
```

Comment:
Exactly. I would move each simplified case into each `Class#from_yaml`, then have `String#to_yaml` implement a complete check for reserved scalar values.